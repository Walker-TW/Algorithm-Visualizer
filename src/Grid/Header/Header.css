.big {
  transform: scale(2);
}
