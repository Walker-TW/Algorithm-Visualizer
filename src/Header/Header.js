import React, { Component } from 'react';
import { Button } from 'react-bootstrap';
import PropTypes from 'prop-types';

export class Header extends Component {
  testDijkstra = (startNode, endNode) => {
    console.log(this.props);
    console.log(startNode, 'Look a start node');
    console.log(endNode, 'Look an end node');
  };

  render() {
    const { startNode, endNode } = this.props;
    return (
      <Button
        onClick={() => this.testDijkstra(startNode, endNode)}
        variant="primary"
      >
        Let's Run Dijkstra
      </Button>
    );
  }
}

export default Header;

Header.propTypes = {
  startNode: PropTypes.object.isRequired,
  endNode: PropTypes.object.isRequired,
};
