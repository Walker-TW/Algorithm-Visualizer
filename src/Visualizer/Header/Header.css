.big {
  transform: scale(2);
}
.btn-run {
  background-color: yellow;
  margin: 0;
  float: none;
}

.nav-center {
  position: absolute;
  left: 50%;
  transform: translatex(-50%);
}

.slider.slider-horizontal {
  width: 70%;
}

.github-img {
  width: 70%;
  height: auto;
  margin: 5px 20% 5px 50%;
}
.linkedin-img {
  width: 90%;
  height: auto;
  margin: 5px 20% 5px 50%;
}
.image-link {
  width: 30%;
}
