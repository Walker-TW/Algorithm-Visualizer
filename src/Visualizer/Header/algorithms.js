export const algorithmDropdowns = [
  {
    id: 'dijkstra',
    name: 'Dijkstra',
    header: 'Weighted',
    content:
      'Running Time:\nO( | E | log | V | )\nAlways finds the shortest path.',
    footer: 'Uses distance of nodes to choose the direction to travel.',
  },
  {
    id: 'astar-e',
    name: 'A* Euclidean',
    header: 'Weighted',
    content:
      'Running Time:\nO( log ( h* )( x ) )\nWill not always find the shortest path.',
    footer: 'Uses as the crow flies heuristic to decide direction of search.',
  },
  {
    id: 'astar-m',
    name: 'A* Manhatten',
    header: 'Weighted',
    content: 'Running Time:\nO( log ( h* )( x ) )\nWill find the shortest path',
    footer: 'Uses the “taxi cab” heuristic for non diagonal graphs.',
  },
  {
    id: 'dfs',
    name: 'Depth First Search',
    header: 'Not Weighted',
    content:
      'Running Time:\nO( | V | + | E | )\nWill not find the shortest path',
    footer: 'Searches every branch of a graph',
  },
  {
    id: 'bfs',
    name: 'Breadth First Search',
    header: 'Not Weighted',
    content: 'Running Time:\nO( | V | + | E | )\nWill find the shortest path',
    footer:
      'Will search paths only after its current path has been fully explored.',
  },
];

export default algorithmDropdowns;
