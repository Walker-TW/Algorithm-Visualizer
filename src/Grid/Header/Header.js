import React from 'react';
import { Button, Form, Nav, Navbar, NavDropdown } from 'react-bootstrap';
import PropTypes from 'prop-types';

import './Header.css';

const Header = (props) => {
  const {
    algorithm,
    fenceToggle,
    // runAstar,
    runDijkstra,
    reset,
    setAlgorithm,
  } = props;

  return (
    <Navbar collapseOnSelect expand="lg" bg="dark" variant="dark">
      <Navbar.Brand href="#home" children={'Algo-Visualiser'} />
      <Navbar.Toggle aria-controls="responsive-navbar-nav" />
      <Navbar.Collapse id="responsive-navbar-nav">
        <Nav className="mr-auto">
          <Nav.Link href="#features" children={'Features'} />
          <Nav.Link href="#pricing" children={'Pricing'} />
          <NavDropdown title="Algorithms" id="collasible-nav-dropdown">
            <NavDropdown.Item
              onClick={setAlgorithm('dijkstra')}
              children={'Dijkstra'}
            />
            <NavDropdown.Item onClick={setAlgorithm('astar')} children={'A*'} />
            <NavDropdown.Item href="#action/3.3" children={'Something'} />
            <NavDropdown.Divider />
            <NavDropdown.Item href="#action/3.4" children={'Separated link'} />
          </NavDropdown>
          <Form>
            <Form.Check
              type="checkbox"
              id="fence-check"
              name="fences"
              label="Fences"
              style={{ color: 'white' }}
              onChange={fenceToggle}
            />
          </Form>
        </Nav>
        <Nav variant="center">
          <Button
            id="button"
            variant="primary"
            onClick={
              algorithm === 'dijkstra'
                ? runDijkstra()
                : // : 'astar'
                  // ? runAstar()
                  ''
            }
            children={`Let's Run ${algorithm}`}
          />
          <Button
            id="reset-btn"
            style={{ backgroundColor: 'red' }}
            onClick={reset}
            variant="primary"
            children={'Reset'}
          />
        </Nav>
        <Nav>
          <Nav.Link href="#deets">More deets</Nav.Link>
          <Nav.Link eventKey={2} href="#memes">
            Contact
          </Nav.Link>
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  );
};
export default Header;

Header.propTypes = {
  algorithm: PropTypes.string.isRequired,
  // runAstar: PropTypes.func.isRequired,
  runDijkstra: PropTypes.func.isRequired,
  fenceToggle: PropTypes.func.isRequired,
  setAlgorithm: PropTypes.func.isRequired,
  reset: PropTypes.func.isRequired,
};
