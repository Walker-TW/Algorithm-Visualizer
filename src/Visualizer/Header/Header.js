import React, { useState } from 'react';
import algorithmInfo from './algorithms.js';
import { animateScroll as scroll } from 'react-scroll';
import {
  Button,
  ButtonGroup,
  Col,
  Container,
  Dropdown,
  DropdownButton,
  Form,
  FormControl,
  Image,
  Nav,
  Navbar,
  Modal,
  NavDropdown,
  OverlayTrigger,
  Popover,
  Row,
} from 'react-bootstrap';
import PropTypes from 'prop-types';
import { getDimensions, getMax } from '../../Helpers/getDimensions';
import './Header.css';

const Header = (props) => {
  const {
    algorithm,
    animationSpeed,
    device,
    fenceToggle,
    ready,
    resizeGrid,
    resetFences,
    resetVisited,
    run: propRun,
    setAlgorithm,
    speed: propsSpeed,
    setNodeSize: propsNodeSize,
  } = props;

  const [screenWidth, screenHeight] = getDimensions();
  const [maxWidth, maxHeight] = getMax();
  const [width, setWidth] = useState(Math.ceil(screenWidth));
  const [height, setHeight] = useState(Math.ceil(screenHeight));
  const [speed, setSpeed] = useState(propsSpeed);
  const [nodeSize, setNodeSize] = useState();

  const [show, setShow] = useState();
  const [expanded, setExpanded] = useState(false);

  const mobile = device === 'mobile';
  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);
  const collapseNav = () => setExpanded(false);

  const run = () => {
    if (algorithm === '') {
      alert('Please select an algorithm');
    }
    if (!ready) {
      alert(
        'Please choose a start and finish point before running by clicking on the desired squares'
      );
    }
    if (ready && algorithm) {
      propRun();
    }
  };

  const gitHubImage = (
    <Image
      src="/images/github.png"
      className={'github-img'}
      alt={'Github'}
      fluid
    />
  );

  const linkedInImage = (
    <Image
      src="/images/linkedin.png"
      alt={'LinkedIn'}
      className={'linkedin-img'}
      fluid
    />
  );

  const algorithmDropdowns = algorithmInfo.map((alg, i) => {
    return (
      <OverlayTrigger
        key={i}
        trigger={['hover', 'focus']}
        placement={mobile ? 'bottom' : 'right'}
        overlay={
          <Popover id={`${alg.id}-popover`}>
            <Popover.Title as="h2" children={`${alg.name}`} />
            <Popover.Content>
              <p>
                <strong>{`${alg.header}`}</strong>
              </p>
              <p style={{ whiteSpace: 'pre-line' }}>{`${alg.content}`}</p>
              <p>{`${alg.footer}`}</p>
            </Popover.Content>
          </Popover>
        }
      >
        <NavDropdown.Item
          id={`set-${alg.id}`}
          onClick={() => setAlgorithm(`${alg.name}`)}
          children={`${alg.name}`}
          active={algorithm === `${alg.name}`}
        />
      </OverlayTrigger>
    );
  });

  const howToUse = (
    <Modal show={show} onHide={handleClose}>
      <Modal.Header closeButton>
        <Modal.Title>How To Use</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <Container>
          <Row>
            1. Place a start and end point by clicking on the grid! (You can
            remove them by clicking on them again)
          </Row>
          <Row>
            {' '}
            2. Then place fences by checking "Fence Mode" and clicking on the
            grid.{' '}
          </Row>
          <Row>3. Choose an algorithm via the "Algorithms" dropdown. </Row>
          <Row>4. Run it via pressing the green "Run Algorithm" button. </Row>
          <Row>5. Enjoy!</Row>
        </Container>
      </Modal.Body>
      <Modal.Footer>
        <Button variant="primary" onClick={handleClose}>
          Close
        </Button>
      </Modal.Footer>
    </Modal>
  );

  const settings = (
    <DropdownButton title="Settings" size="sm" variant="dark">
      <Container variant="dark">
        <Row>
          <Form variant="dark" inline>
            <Col>
              Node Size
              <FormControl
                size="sm"
                type="text"
                placeholder={`Currently ${nodeSize})`}
                onChange={(e) => {
                  setNodeSize(e.target.value);
                }}
              />
              <Form.Control
                type="range"
                size="sm"
                min="1"
                max="100"
                value={nodeSize}
                onChange={(e) => {
                  setNodeSize(e.target.value);
                  propsNodeSize(e.target.value);
                }}
                custom
              />
              Grid Size
              <FormControl
                size="sm"
                type="text"
                placeholder={`Width (Currently ${width})`}
                onChange={(e) => {
                  setWidth(e.target.value);
                }}
              />
              <Form.Control
                type="range"
                size="sm"
                min="1"
                max={maxWidth}
                value={width}
                onChange={(e) => {
                  setWidth(e.target.value);
                  resizeGrid([e.target.value, height]);
                }}
                custom
              />
              <NavDropdown.Divider />
              <FormControl
                type="text"
                size="sm"
                placeholder={`Height (Currently ${height})`}
                onChange={(e) => {
                  setHeight(e.target.value);
                }}
                className="Row-Input"
              />
              <Form.Control
                type="range"
                min="1"
                max={maxHeight}
                value={height}
                onChange={(e) => {
                  setHeight(e.target.value);
                  resizeGrid([width, e.target.value]);
                }}
                custom
              />
              <NavDropdown.Divider />
              <Form.Label children={'Draw Square'} />
              <Form.Control
                type="range"
                size="sm"
                min="1"
                max="50"
                value={(height, width)}
                onChange={(e) => {
                  setWidth(e.target.value);
                  setHeight(e.target.value);
                  resizeGrid([e.target.value, e.target.value]);
                }}
                custom
              />
              <NavDropdown.Divider />
              <Form.Label children={'Animation Speed'} />
              <Form.Control
                type="range"
                min="1"
                max="5"
                value={speed}
                onChange={(e) => {
                  setSpeed(e.target.value);
                  animationSpeed(e.target.value);
                }}
                custom
              />
            </Col>
          </Form>
        </Row>
      </Container>
    </DropdownButton>
  );

  const contactInfo = (
    <DropdownButton title="Contact The Team" size="sm" id="contact-info">
      <Container>
        <Row>
          <NavDropdown.Item id={'bassel'} children={'Bassel'} />
          <a
            className={'image-link'}
            href="https://github.com/basselalsayed"
            children={gitHubImage}
          />
          <a
            className={'image-link'}
            href="https://www.linkedin.com/in/bsas/"
            children={linkedInImage}
          />
        </Row>
        <Row>
          <NavDropdown.Item id={'tom'} children={'Tom'} />
          <a
            className={'image-link'}
            href="https://github.com/Walker-TW"
            children={gitHubImage}
          />
          <a
            className={'image-link'}
            href="https://www.linkedin.com/in/thomas-w-walker"
            children={linkedInImage}
          />
        </Row>
      </Container>
    </DropdownButton>
  );

  return (
    <Navbar
      expanded={expanded}
      expand="lg"
      bg="dark"
      variant="dark"
      collapseOnSelect
    >
      <Navbar.Brand
        href="https://github.com/Walker-TW/Algorithm-Visualizer"
        children={'Algo-Visualiser'}
      />
      <Navbar.Toggle
        onClick={() => setExpanded(expanded ? false : 'lg')}
        aria-controls="responsive-navbar-nav"
      />
      <Navbar.Collapse id="responsive-navbar-nav">
        <Container fluid>
          <Col md={{ span: 2 }}>
            <Nav className="mr-auto">
              <NavDropdown title="Algorithms" id="collapsible-nav-dropdown">
                {algorithmDropdowns}
              </NavDropdown>
            </Nav>
          </Col>
          <Col md={{ span: 4 }}>
            <Nav>
              <Button
                id="info-btn"
                style={{ border: '2px solid cyan', color: 'cyan' }}
                variant="dark"
                children={'How To Use'}
                onClick={handleShow}
              />
              {howToUse}
              <Button
                id="run-btn"
                style={{ border: '2px solid chartreuse', color: 'chartreuse' }}
                variant="dark"
                onClick={
                  mobile
                    ? () => {
                        setTimeout(() => run(), 1000);
                        collapseNav();
                        scroll.scrollToBottom({
                          duration: 1500,
                          delay: 100,
                          smooth: true,
                        });
                      }
                    : () => setTimeout(() => run(), 500)
                }
                children={
                  algorithm
                    ? `Let's Run ${algorithm}`
                    : 'Please Select Algorithm'
                }
                disabled={!ready || algorithm === ''}
              />
              <Dropdown as={ButtonGroup}>
                <Button
                  id="reset-btn"
                  variant="dark"
                  style={{ border: '2px solid red', color: 'red' }}
                  children={'Reset Visited'}
                  onClick={resetVisited}
                />
                <Dropdown.Toggle
                  split
                  variant="dark"
                  style={{ border: '2px solid red', color: 'red' }}
                  id="dropdown-custom-2"
                />
                <Dropdown.Menu>
                  <Dropdown.Item
                    id="fence-reset-btn"
                    onClick={resetFences}
                    variant="dark"
                    children={'Reset Fences'}
                  />
                </Dropdown.Menu>
              </Dropdown>
            </Nav>
          </Col>
          <Col md={{ span: 4 }}>
            <Nav navbar="true">
              <Container>
                <Form inline>
                  <Form.Check
                    type="switch"
                    id="fence-check"
                    name="fences"
                    label="Fence mode"
                    style={{ color: 'white' }}
                    onChange={fenceToggle}
                  />
                </Form>
              </Container>
              {settings}
              {contactInfo}
            </Nav>
          </Col>
        </Container>
      </Navbar.Collapse>
    </Navbar>
  );
};
export default Header;

Header.propTypes = {
  algorithm: PropTypes.string.isRequired,
  animationSpeed: PropTypes.func.isRequired,
  device: PropTypes.string.isRequired,
  fenceToggle: PropTypes.func.isRequired,
  ready: PropTypes.bool.isRequired,
  resetFences: PropTypes.func.isRequired,
  resetVisited: PropTypes.func.isRequired,
  resizeGrid: PropTypes.func.isRequired,
  run: PropTypes.func.isRequired,
  setAlgorithm: PropTypes.func.isRequired,
  speed: PropTypes.string.isRequired,
};
