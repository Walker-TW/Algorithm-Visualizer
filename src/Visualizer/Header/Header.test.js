import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';

import Header from './Header';

const runSpy = jest.fn(() => {});
const fenceToggleSpy = jest.fn(() => {});
const resetSpy = jest.fn(() => {});
const setAlgorithmSpy = jest.fn(() => {});
const mazeBuildSpy = jest.fn(() => {});

const defaultProps = {
  algorithm: '',
  run: runSpy,
  fenceToggle: fenceToggleSpy,
  reset: resetSpy,
  setAlgorithm: setAlgorithmSpy,
  ready: false,
  mazeBuild: mazeBuildSpy,
};

const readyProps = {
  ...defaultProps,
  algorithm: 'dijkstra',
  ready: true,
};

describe('<Header />', () => {
  const wrapper = shallow(<Header {...defaultProps} />);

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls the run spy onclick', () => {
    const jsdomAlert = window.alert;
    window.alert = jest.fn();

    const wrapper = mount(<Header {...readyProps} />);
    wrapper.find('#run-btn').last().simulate('click');

    expect(runSpy.mock.calls.length).toEqual(1);
    window.alert = jsdomAlert;
  });

  it('calls the wall toggle spy onclick', () => {
    wrapper
      .find('#fence-check')
      .simulate('change', { target: { checked: true } });
    expect(fenceToggleSpy.mock.calls.length).toEqual(1);
  });

  it('calls the reset spy onclick', () => {
    wrapper.find('#reset-btn').simulate('click');
    expect(resetSpy.mock.calls.length).toEqual(1);
  });

  it('calls the set algorithm spy onclick', () => {
    wrapper.find('#set-dijkstra').simulate('click');
    wrapper.find('#set-astar-manhatten').simulate('click');
    wrapper.find('#set-astar-euclidean').simulate('click');
    expect(setAlgorithmSpy.mock.calls.length).toEqual(3);
  });
  it('calls the maze building spy onclick', () => {
    wrapper.find('#maze-btn').simulate('click');
    expect(mazeBuildSpy.mock.calls.length).toEqual(1);
  });
});
