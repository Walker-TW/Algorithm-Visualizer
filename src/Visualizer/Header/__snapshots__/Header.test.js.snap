// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders 1`] = `
<Navbar
  bg="dark"
  collapseOnSelect={true}
  expand="lg"
  variant="dark"
>
  <NavbarBrand
    href="https://github.com/Walker-TW/Algorithm-Visualizer"
  >
    Algo-Visualiser
  </NavbarBrand>
  <NavbarToggle
    aria-controls="responsive-navbar-nav"
    label="Toggle navigation"
  />
  <NavbarCollapse
    id="responsive-navbar-nav"
  >
    <Container
      fluid={true}
    >
      <Col>
        <Nav
          className="mr-auto"
          fill={false}
          justify={false}
        >
          <NavDropdown
            title="Mazes"
          >
            <DropdownItem
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
            >
              1
            </DropdownItem>
            <DropdownDivider
              role="separator"
            />
            <DropdownItem
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
            >
              2
            </DropdownItem>
            <DropdownDivider
              role="separator"
            />
            <DropdownItem
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
            >
              3?
            </DropdownItem>
            <DropdownDivider
              role="separator"
            />
          </NavDropdown>
          <NavDropdown
            id="collasible-nav-dropdown"
            title="Algorithms"
          >
            <DropdownItem
              active={false}
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
              id="set-dijkstra"
              onClick={[Function]}
            >
              Dijkstra
            </DropdownItem>
            <DropdownItem
              active={false}
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
              id="set-astar-euclidean"
              onClick={[Function]}
            >
              A* (Euclidean Distance)
            </DropdownItem>
            <DropdownItem
              active={false}
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
              id="set-astar-manhatten"
              onClick={[Function]}
            >
              A* (Manhatten Distance)
            </DropdownItem>
            <DropdownItem
              active={false}
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
              id="set-depth-first-search"
              onClick={[Function]}
            >
              Depth First Search
            </DropdownItem>
            <DropdownItem
              active={false}
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
              id="set-breadth-first-search"
              onClick={[Function]}
            >
              Breadth First Search
            </DropdownItem>
          </NavDropdown>
        </Nav>
      </Col>
      <Col>
        <Nav
          fill={false}
          justify={false}
        >
          <Button
            active={false}
            disabled={true}
            id="run-btn"
            onClick={[Function]}
            style={
              Object {
                "border": "2px solid chartreuse",
                "color": "chartreuse",
              }
            }
            type="button"
            variant="dark"
          >
            Please Select An Algorithm
          </Button>
          <SplitButton
            id="reset-btn"
            onClick={[MockFunction]}
            style={
              Object {
                "border": "2px solid red",
                "color": "red",
              }
            }
            title="Reset Visited"
            toggleLabel="Toggle dropdown"
            type="button"
            variant="dark"
          >
            <DropdownItem
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
              id="start-reset"
              onClick={[Function]}
              variant="dark"
            >
              Reset Start
            </DropdownItem>
            <DropdownItem
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
              id="fence-reset-btn"
              onClick={[MockFunction]}
              variant="dark"
            >
              Reset Fences
            </DropdownItem>
            <DropdownItem
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
              id="end-reset"
              onClick={[Function]}
              variant="dark"
            >
              Reset End
            </DropdownItem>
          </SplitButton>
        </Nav>
      </Col>
      <Col>
        <Nav
          fill={false}
          justify={false}
          navbar="true"
        >
          <Container
            fluid={false}
          >
            <Form
              inline={true}
            >
              <FormCheck
                disabled={false}
                id="fence-check"
                inline={false}
                isInvalid={false}
                isValid={false}
                label="Fence mode"
                name="fences"
                onChange={[MockFunction]}
                style={
                  Object {
                    "color": "white",
                  }
                }
                title=""
                type="switch"
              />
            </Form>
          </Container>
          <DropdownButton
            size="sm"
            title="Settings"
            variant="dark"
          >
            <Container
              fluid={false}
              variant="dark"
            >
              <Row
                noGutters={false}
              >
                <Form
                  inline={true}
                  variant="dark"
                >
                  <Col>
                    Grid Size
                    <FormControl
                      onChange={[Function]}
                      placeholder="Width (Currently 30)"
                      size="sm"
                      type="text"
                    />
                    <FormControl
                      custom={true}
                      max={34.13333333333333}
                      min="1"
                      onChange={[Function]}
                      size="sm"
                      type="range"
                      value={30}
                    />
                    <DropdownDivider
                      role="separator"
                    />
                    <FormControl
                      className="Row-Input"
                      onChange={[Function]}
                      placeholder="Height (Currently 15)"
                      size="sm"
                      type="text"
                    />
                    <FormControl
                      custom={true}
                      max="50"
                      min="1"
                      onChange={[Function]}
                      type="range"
                      value={15}
                    />
                    <DropdownDivider
                      role="separator"
                    />
                    <FormLabel
                      column={false}
                      srOnly={false}
                    >
                      Draw Square
                    </FormLabel>
                    <FormControl
                      custom={true}
                      max="50"
                      min="1"
                      onChange={[Function]}
                      size="sm"
                      type="range"
                      value={30}
                    />
                    <DropdownDivider
                      role="separator"
                    />
                    <FormLabel
                      column={false}
                      srOnly={false}
                    >
                      Animation Speed
                    </FormLabel>
                    <FormControl
                      custom={true}
                      max="5"
                      min="1"
                      onChange={[Function]}
                      type="range"
                    />
                  </Col>
                </Form>
              </Row>
            </Container>
          </DropdownButton>
          <DropdownButton
            id="contact-info"
            size="sm"
            title="Contact The Team"
          >
            <Container
              fluid={false}
            >
              <Row
                noGutters={false}
              >
                <DropdownItem
                  as={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "displayName": "SafeAnchor",
                      "render": [Function],
                    }
                  }
                  disabled={false}
                  href="https://github.com/basselalsayed"
                  id="basselGit"
                >
                  Bassel's Git
                </DropdownItem>
              </Row>
              <Row
                noGutters={false}
              >
                <DropdownItem
                  as={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "displayName": "SafeAnchor",
                      "render": [Function],
                    }
                  }
                  disabled={false}
                  href="https://www.linkedin.com/in/bsas/"
                  id="basselLink"
                >
                  Bassel's LinkedIn
                </DropdownItem>
              </Row>
            </Container>
            <Container
              fluid={false}
            >
              <Row
                noGutters={false}
              >
                <DropdownItem
                  as={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "displayName": "SafeAnchor",
                      "render": [Function],
                    }
                  }
                  disabled={false}
                  href="https://github.com/Walker-TW"
                  id="tomGit"
                >
                  Tom's Git
                </DropdownItem>
              </Row>
              <Row
                noGutters={false}
              >
                <DropdownItem
                  as={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "displayName": "SafeAnchor",
                      "render": [Function],
                    }
                  }
                  disabled={false}
                  href="https://www.linkedin.com/in/thomas-w-walker/"
                  id="tomLink"
                >
                  Tom's LinkedIn
                </DropdownItem>
              </Row>
            </Container>
          </DropdownButton>
        </Nav>
      </Col>
    </Container>
  </NavbarCollapse>
</Navbar>
`;
